### ISSUE TITLE
possibility to 'reconfigure' entities

### ISSUE THREAD

**Issue by krismarc** [alengwenus/lcn-frontend#36]
Hi @alengwenus 

I just made my one of my existing led lights dimmable by replacing it's power supply to dimmable and I've faced a problem.

In yaml way, I could actually see current configuration details eg. transition for the output dimming.
In new frontend, I can't see actual value and recreation is only possible by removing and creating over entity and once re-created I can't lookup details again.

..and if I am not wrong, even if you use the same name, entities are nolonger the same so dashboards and all other things become broken.

Best regards,
K.M.

**Comment by alengwenus** [comment]
Checking the reconfiguration is on my todo list. However, as long as you  set the same entity_id for the new entity your dashboard shouldn't break.

**Comment by crosenbe** [comment]
I started to work on that but i mentioned why it is a bit more complex. The path is designed to work one-way. The config item does not contain the sourcetype. I have also seen the path is starting with the domain dialog, next sourcetype and then down to the details. The mappings are sometimes in different dialos if they can be used in different domains. 

One idea is to centralize the lcn stuff to lcn.ts and extend it by more details, i.e.

```
[
      { domain: "light", souretype: "output", name: output + " 1", value: "OUTPUT1" },
      { domain: "light", souretype: "output", name: output + " 2", value: "OUTPUT2" },
      { domain: "light", souretype: "output", name: output + " 3", value: "OUTPUT3" },
      { domain: "light", souretype: "output", name: output + " 4", value: "OUTPUT4" },
];
```

Provide a function to obain the values in the relevant dialogs and store the sourcetype to the domain_data to make it more easy to reconstruct the dialog in editing mode.

Alternatively it's possible search in all ConfigItem definitions and calculate the sourcetype to rebuild the dialog.

**Comment by crosenbe** [comment]
I worked on a first proposal, it's not complete because i implemented only the domain `light`. Because i didn't find an entity management it's not simple to update individual entries in runtime, maybe it makes sense to include a dict to identify and update entities in runtime.

**ha-core**
https://github.com/home-assistant/core/compare/dev...crosenbe:homeassistant-core:lcn_update_entities

**lcn-frontend**
https://github.com/alengwenus/lcn-frontend/compare/main...crosenbe:lcn-frontend:edit_entity

i still think about moving the multiple used data structures from the domain related ts files into const.ts.

**Comment by alengwenus** [comment]
Hi @crosenbe,
I had a quick look at your changes. Withouth testing it I think that's the way to go and it's in line with my initial work in this area.
Unfortunately it's not possible to update individual entities at runtime. Therefore, the only option left is the method you already used: reloading the ConfigEntry completely.

You're welcome to keep working on it, but at the moment I'm not sure if I'll be developing any new features for the front-end configuration dashboard myself. The main reasons for this are that it takes a lot of ongoing work to keep the frontend up to date with the latest HA versions. Also, PR approvals in HA Core simply take too long for me to be able to work on features continuously.

I've been exploring other approaches for quite some time now:
First I evaluated the [Config SubEntries](https://developers.home-assistant.io/blog/2025/02/16/config-subentries/) to avoid having to program the front end by myself and to use only the HA functions provided natively. The MQTT integration is one example which heavily relies on the Config Subentries. However, the configuration options for LCN are extremely complex, so this approach is also very confusing and not very user-friendly.

Second (and that's what I believe will be the long-term solution for me) is a new project started by me: [lcn2mqtt](https://github.com/alengwenus/lcn2mqtt).
This project is quite similar to the [Zigbee2MQTT](https://www.zigbee2mqtt.io/) project and just provides a bridge between MQTT and the LCN bus using the same libraries under the hood as the native LCN integration, The great thing about it is, that it benefits from all features of HA's MQTT integration. This includes the MQTT discovery, which is able to (more or less) automatically add your modules including entities to HA on the fly. Or you may choose to configure everything on your own. Even OpenHAB supports HA's MQTT discovery. So you could also use it within this ecosystem. Or you could simply call the respective MQTT commands from any other application.

**Comment by crosenbe** [comment]
Hi @alengwenus,

i completely understand your concerns about developing with Home Assistant and the UI. Both were new to me and it was many times very confusing. A logical consequence is to avoid all development which is not necessary and to focus on the LCN part. I absolutely like your lcn2mqtt approach. Is it in a state for a first look? 

I prefer to give it a try before i spend more time into the edit entity story because i expect i will again invest hours and days with the UI.

**Comment by alengwenus** [comment]
Yes, just try it out. Not every LCN feature is already implemented because I first focused on the architecture. It might also have some variable issues for really old modules. But it will be quite easy to implement the missing features which I plan to do little by little.
Note that there is some documentation in the project's wiki!

**Comment by crosenbe** [comment]
I read it at the moment and will create a Helm Chart. As far i understood the configuraiton of the devices is done with configuration.yaml. Do you plan to extend it with an API, i.e. FastAPI to provide a couple of endpoints to add/edit/remove devices and configurations? This means a new device or configuration will lead to a change in this file and a restart of the container.

**Comment by alengwenus** [comment]
It's not yet on my todo list. Let's welcome back the frontend stuff. 😉
Maybe we can vibe code something without to much effort.

**Comment by crosenbe** [comment]
Great job done, my first pod is up and running. I added a bit to the code because some people like me use websockets instead of mqtt native transport. Which makes much more sense if the container is working behind an ingress. Also also found some missing __future__ annotations imports.

https://github.com/alengwenus/lcn2mqtt/compare/dev...crosenbe:lcn2mqtt:websocket
